openapi-style-guide.md

```diff
@@ -0,0 +1,208 @@
+For services that use date-based versions, this should be a date in YYYY-MM-DD format, optionally suffixed with '-preview'.
+
+<!-- --------------------------------------------------------------- -->
+
+## Operations
+
+### Summary and description
+
+Each operation should have a summary and/or description.
+When both a summary and description are provided, the description should include additional detail
+about the operation behavior -- it should not simply restate the summary.
+
+### OperationId
+
+Every operation should have a unique `operationId`.
+
+The `operationId` should be of the form `Noun_Verb`.  It should contain exactly one underscore.
+
+AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the Verb becomes the name of the method in that class, i.e., operations are grouped inside the operations class named after the noun.
+
+The `Verb` of the `operationId` should be or contain a specific value depending on the operation method:
+
+| operation method | verb should contain | notes  |
+| ---------------- | ------------------- | ------ |
+| get              | "Get" or "List"     | should be "List" if response is pageable |
+| put              | "Create" or "Update" | could be "CreateOrUpdate" |
+| patch            | "Update"            | could be "CreateOrUpdate" |
+| delete           | "Delete"            | |
+
+Further, a put or patch that supports both create and update should have "Create" and "Update" in the verb.
+
+### Request body
+
+The request body for a patch operation should be `application/merge-patch+json` content type.
+
+### Response body
+
+All success responses except 202 and 204 should define a response body.
+
+Responses for status codes 202 and 204 should have no response body.
+
+A delete operation should have a 204 response.
+
+For a path with a "create" operation (put or patch that returns 201),
+the 200 response of get, put, and patch, if present, should have the same response body schema
+as the create operation 201 response.
+
+### Support for pagination
+
+To support pagination:
+- The operation should have the `x-ms-pageable` annotation
+- The operation response should contain a top-level `value` property of type array and required
+- The operation response should contain a top-level `nextLink` property of type string and optional
+- If the operation has a `skip` parameter, it must be an integer and optional
+- If the operation has a `top` parameter, it must be an integer, optional, and have a documented default and maximum value
+- If the operation has a `maxpagesize` parameter, it must be an integer, optional, and have a documented default and maximum value
+
+### Error response
+
+All operations should have a "default" (error) response with a response body that conforms to the Azure API Guidelines.
+
+All `4xx` and `5xx` responses should specify `x-ms-error-response: true` except for `404` response of HEAD operation.
```

Comment:
Why should people do that?  Isn't it implicitly true for anything 4XX or above?

Reply:
To be honest I don't really understand the need / value of this extension. I had originally thought that it was only needed for the `default` response, since it could be either an error response or an expected response, but then someone (I don't remember who) told me that Autorest always treats `default` as an error response so the annotation isn't needed there.  I guess I need to do some more research on this one.

Reply:
Yeah, you don't fully understand because we haven't really documented anything at https://azure.github.io/autorest/extensions/#x-ms-error-response.  I had to go look it up and apparently I was wrong with the latest version of AutoRest: https://github.com/Azure/autorest/blob/main/packages/extensions/modelerfour/src/modeler/modelerfour.ts#L2067  

Please ignore my comment.  :smile: